lib/mongo/database/view.rb

```diff
@@ -164,8 +180,12 @@ def collections_info_spec(session, options = {})
           db_name: @database.name,
           session: session
         }.tap do |spec|
-          spec[:selector][:nameOnly] = true if options[:name_only]
+          # nameOnly and authorizedCollections default to false
+          if options[:name_only] ? (spec[:selector][:nameOnly] = true) : (spec[:selector][:nameOnly] = false)
```

Comment:
We do not need to specify false values. Was there some case that the existing code didn't handle? It seems sufficient to me.

Reply:
If a user were to pass in name_only as false into list_collections (which does not force name_only to be true), then nameOnly would be set to nil because spec[:selector][:nameOnly] = true only when options[name_only] is true.